ImportExport/ObjectsToPython/ObjectsToPython.FCMacro

```diff
@@ -0,0 +1,344 @@
+            addthis.append(obj.Source)
+        if hasattr(obj, 'Links'):
+            addthis += obj.Links
+            
+    added = 0
+    for a in addthis:
+        if (not a in selection) and hasattr(obj, 'Name') and hasattr(obj, 'TypeId'):
+            selection.insert(0, a)
+            added = added + 1
+            
+    if added > 0:
+        expandSelection(selection)
+            
+
+def addObject(doc, obj, objectlist): 
+    objname = varname(obj)
+    addScript("%s = doc.addObject('%s', '%s')"%(objname, obj.TypeId, objname))
+    defaultobj = doc.addObject(obj.TypeId, obj.Name + "Default")
+    addProperties(obj, objname, defaultobj, objectlist)
+    addProperties(obj.ViewObject, objname + ".ViewObject", defaultobj.ViewObject, [])                      
+    doc.removeObject(obj.Name + "Default")      
+    
+    
+def addProperties(obj, objname, refobj, objectlist):
+    skipProperties = ["Label", "Shape", "Proxy", "AddSubShape"]
+    skipObjProp = [("Sketcher::SketchObject", "Geometry"), ("Sketcher::SketchObject", "Constraints"),
+        ('PartDesign::Body', 'Origin'), ('PartDesign::Body', 'Tip')]
+    
+    if obj.TypeId == "Spreadsheet::Sheet":
+        addSpreadsheet(obj, objname)
+        return
+    
+    if obj.TypeId == "Sketcher::SketchObject":
+        addSketch(obj, objname)
+
+    for propname in obj.PropertiesList:
+        if not propname in skipProperties and not (obj.TypeId, propname) in skipObjProp:
+            prop = obj.getPropertyByName(propname)  
+            val = objectToText(prop, objectlist)
+            
+            try:
+                refprop = refobj.getPropertyByName(propname)
+                defaultval = objectToText(refprop, objectlist)
+            except:
+                defaultval = None       
+                
+            if propname == "Support":
+                val = val.replace("XY_Plane", "doc.XY_Plane")
+                val = val.replace("XZ_Plane", "doc.XZ_Plane")
+                val = val.replace("YZ_Plane", "doc.YZ_Plane")
+                
+            if propname == "ExpressionEngine":
+                for expression in prop:
+                    addScript("%s.setExpression%s"%(objname, expression))  
+                    
+            elif propname == "_Body":
+                addScript("%s_Group.append(%s)"%(val, objname))
+                          
+            elif (obj.TypeId, propname) == ('PartDesign::Body', 'Group'):
+                addScript("%s_Group = []"%(objname))
+                                                                                     
+            elif objectToText(prop, objectlist) is not None:   
+                if val != defaultval:        
+                    addScript("%s.%s = %s"%(objname, propname, val))   
+                                      
+                                                                                                                            
+def objectToText(obj, objectlist = []):    
```

Comment:
`objectlist` will not be reinitialized to an empty list on further calls, is this OK?

Reply:
this is an optional parameter with default initialization.

Reply:
What I mean is that if you change `objectlist` in the first call to `objectToText`, `objectlist` will be modified also for the consecutive calls to `objectToText`. Try by yourself:
```python
def f(l=[]): 
    l.append(len(l)) 
    print(l) 
f()
f()
```

Reply:
in this case, objectlist will not be modified in objectToText()